src/resources.js

```diff
@@ -1307,6 +1319,37 @@ export class Resource {
       this.state_ = ResourceState_.NOT_LAID_OUT;
     }
   }
+
+  /**
+   * Only allowed in dev mode when runtime is turned off. Performs all steps
```

Comment:
maybe we can add an `if (dev)` block around the whole code here so it can be DCE'd?